### PR TITLE
Pagination 1

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes
Adds ic-pagination and ic-pagination-item components

## Related issue
#481 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Review by MI6-451 — APPROVED** [review]

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]
Just two more tiny things - otherwise looks great :)

File: packages/web-components/src/components/ic-pagination-item/ic-pagination-item.css [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  text-align: center;
+}
+
+.item-container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: calc(var(--ic-space-xl) + var(--ic-space-sm));
+  height: var(--ic-space-xl);
+  margin: 4px var(--ic-space-xxxs);
```
Comment (GCHQ-Developer-530) [inline packages/web-components/src/components/ic-pagination-item/ic-pagination-item.css:13]: Can this 4px be updated to a token?

File: packages/react/src/stories/ic-pagination.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    <IcPagination pages="15" type="complex" disabled />
+  </Story>
+</Canvas>
+
+### Complex - Adjacent & Boundary Items Set
+
+<Canvas>
+  <Story name="Complex - Adjacent & Boundary Items Set">
+    <div>
+      <p style={{ padding: 10 }}>Adacent: 0, Boundary: 0</p>
```
Comment (GCHQ-Developer-530) [inline packages/react/src/stories/ic-pagination.stories.mdx:73]:
```suggestion
      <p style={{ padding: 10 }}>Adjacent: 0, Boundary: 0</p>
```
I think this line has been copy and pasted for the rest so will need updating for those too

**Comment by GCHQ-Developer-847** [comment]
> High-contrast mode doesn't display the selected page

Looks like this is still an issue (everything else looks good to me though)

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**